src/Mimi.jl

```diff
@@ -216,14 +216,15 @@ function setparameter(m::Model, component::Symbol, name::Symbol, value)
 end
 
 function checklabels(m::Model, component::Symbol, name::Symbol, parametername::Symbol, p::ArrayModelParameter)
-    if !(eltype(p.values) <: getmetainfo(m, component).parameters[parametername].datatype)
+    meta_comp_name = Symbol(string(m.components2[component].component_type))
+    if !(eltype(p.values) <: getmetainfo(m, meta_comp_name).parameters[parametername].datatype)
```

Comment:
The error is slightly different depending on whether or not I give the adder component a different name.

When I do:
`addcomponent(model, adder)`
`setparameter(model, :adder, :input, x)`

The error is "KeyError: key :adder not found"

Whereas when I do 

`addcomponent(model, adder, :compA)`
`setparameter(model, :compA, :input, x)`

The error is "KeyError: key (:Mimi,Symbol("Mimi.adder")) not found"